containers/payments/subscription/BlackFridayModal.js

```diff
@@ -0,0 +1,218 @@
+    Price
+} from 'react-components';
+import { checkSubscription } from 'proton-shared/lib/api/payments';
+import { CYCLE, DEFAULT_CURRENCY, DEFAULT_CYCLE } from 'proton-shared/lib/constants';
+import { c } from 'ttag';
+
+const { MONTHLY, YEARLY, TWO_YEARS } = CYCLE;
+
+const BlackFridayModal = ({ bundles = [], onSelect, ...rest }) => {
+    const api = useApi();
+    const [loading, withLoading] = useLoading();
+    const [currency, updateCurrency] = useState(DEFAULT_CURRENCY);
+    const [pricing, updatePricing] = useState({});
+
+    const DEAL_TITLE = {
+        [MONTHLY]: c('Title').t`1 month deal`,
+        [YEARLY]: c('Title').t`1 year deal`,
+        [TWO_YEARS]: c('Title').t`2 year deal`
+    };
+
+    const BILLED_DESCRIPTION = ({ cycle, amount, notice }) =>
+        ({
+            [MONTHLY]: c('Title').jt`Billed as ${amount} (${notice})`,
+            [YEARLY]: c('Title').jt`Billed as ${amount} (${notice})`,
+            [TWO_YEARS]: c('Title').jt`Billed as ${amount} (${notice})`
+        }[cycle]);
+
+    const AFTER_INFO = ({ amount, notice }) =>
+        ({
+            1: c('Title')
+                .jt`(${notice}) Renews after 1 year at a discounted annual price of ${amount} per year (20% discount).`,
+            2: c('Title')
+                .jt`(${notice}) Renews after 2 years at a discounted 2-year price of ${amount} per year (33% discount).`,
+            3: c('Title')
+                .jt`(${notice}) Renews after 2 years at a discounted 2-year & bundle price of ${amount} per year (49% discount).`
+        }[notice]);
+
+    const getBundlePrices = async () => {
+        const result = await Promise.all(
+            bundles.map(({ planIDs = [], cycle = DEFAULT_CYCLE, couponCode }) => {
+                return Promise.all([
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            CouponCode: couponCode,
+                            Currency: currency,
+                            Cycle: cycle
+                        })
+                    ),
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            Currency: currency,
+                            Cycle: cycle
+                        })
+                    ),
+                    api(
+                        checkSubscription({
+                            PlanIDs: planIDs,
+                            Currency: currency,
+                            Cycle: MONTHLY
+                        })
+                    )
+                ]);
+            })
+        );
+
+        updatePricing(
+            result.reduce((acc, [withCoupon, withoutCoupon, withoutCouponMonthly], index) => {
+                acc[index] = {
+                    withCoupon: withCoupon.Amount + withCoupon.CouponDiscount,
+                    withoutCoupon: withoutCoupon.Amount + withoutCoupon.CouponDiscount,
```

Comment:
It sounds a bit strange to have a coupon discount without a coupon `withoutCoupon.CouponDiscount`

Reply:
I know but it's because the `BUNDLE` coupon is coming when the user pick MAIL PLUS and VPN PLUS.